io/plugins/eu.esdihumboldt.hale.io.gml/src/eu/esdihumboldt/hale/io/gml/reader/internal/wfs/WfsBackedGmlInstanceCollection.java

```diff
@@ -228,7 +228,15 @@ public WfsBackedGmlInstanceCollection(LocatableInputSupplier<? extends InputStre
 		// Use primordial URI and issue "hits" request to check if the WFS will
 		// return anything at all
 		int hits;
-		if (ignoreNumberMatched) {
+		boolean isUsedresolvedepth = false;
+		for (NameValuePair nameValuePair : params) {
+			if ("resolvedepth".equalsIgnoreCase(nameValuePair.getName())) {
+				isUsedresolvedepth = true;
+				break;
+			}
+		}
+
+		if (ignoreNumberMatched || isUsedresolvedepth) {
```

Comment:
@emanuelaepure10 I think this can be reduced to a single line as `primordialQueryParams` contains all query parameters in upper case ([see here](https://github.com/halestudio/hale/blob/da47d95df419ab8fd1930ed4b56c12ab55ac0b8e/io/plugins/eu.esdihumboldt.hale.io.gml/src/eu/esdihumboldt/hale/io/gml/reader/internal/wfs/WfsBackedGmlInstanceCollection.java#L198))

```suggestion
		if (ignoreNumberMatched || primordialQueryParams.contains("RESOLVEDEPTH")) {
```

Reply:
Thanks! I overlooked `primordialQueryParams`.